oauth2client/service_account.py

```diff
@@ -683,3 +664,41 @@ def _create_token(self, additional_claims=None):
         jwt = crypt.make_signed_jwt(self._signer, payload,
                                     key_id=self._private_key_id)
         return jwt.decode('ascii'), expiry
+
+    def apply(self, headers, access_token=None):
+        """Add the access token as the Authorization header in headers.
+
+        Args:
+            headers: dict, the headers to add the Authorization header to.
+            access_token: An access token to use instead of the credential's
+                current access token.
+        """
+        if not access_token:
+            access_token = self.access_token
```

Comment:
see the `before_request` logic below, sometimes the credentials need to generate a one-time token for the url. It passes that one-time token into here.